src/Integrations/class-google-web-stories.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Google Web Stories integration class
+ *
+ * @package Parsely\Integrations
+ * @since 3.2.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Integrations;
+
+use Parsely\Parsely;
+
+/**
+ * Integrates Parse.ly tracking with the Google Web Stories plugin.
+ *
+ * @since 3.2.0
+ */
+final class Google_Web_Stories implements Integration {
+	/**
+	 * Instance of Parsely class.
+	 *
+	 * @var Parsely
+	 */
+	private $parsely;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Instance of Parsely class.
+	 */
+	public function __construct( Parsely $parsely ) {
+		$this->parsely = $parsely;
+	}
+
+	/**
+	 * Apply the hooks that integrate the plugin or theme with the Parse.ly plugin.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return void
+	 */
+	public function integrate(): void {
+		if ( defined( 'WEBSTORIES_PLUGIN_FILE' ) ) {
+			add_action( 'web_stories_print_analytics', array( $this, 'render_amp_analytics_tracker' ) );
```

Comment:
Why is the constant needing to be checked? If the plugin isn't active, the hook won't be defined and won't run.

Reply:
I agree. However, I expect that most of our users won't have the plugin, so we'll always be enqueuing an action that it is never executed. I think it has a smaller performance toll to do a `defined` check than an unconditional `add_action`.  We are also doing something similar in the FBIA class.